### PR TITLE
Enhancement: Step link logic

### PR DESCRIPTION
**Pull request by bradsbrown**
Add a plugin endpoint to support adding a link to step text.

Note that I am proposing moving the `url` plugin for tag url parsing over to `tag_url`, BUT keeping nonbreaking legacy support for the `url` key.

### CODE REVIEW

**Comment by dgou** [comment]
Scanning over the code, it all looks very straight forward.
I am curious about including the step keyword though...
```
Given a step is used here
```
```
Given some step comes first
And a step is used here
```

It seems, having not pulled the code yet, that the step URL maker would see those differently?

**Comment by bbrownjc** [comment]
That’s the reason for using `.step_type.capitalize()` rather than `.keyword` — `keyword` can be `And` if that’s the literal value in the Gherkin, but `step_type` will always reliably be in `[“given”, “when”, “then”]`

**Comment by dgou** [comment]
Also thinking out loud, it seems as if there wouldn't be much cause for confusion re: some string being a step or some string being a URL, so one "turn things into links" function.....
It's a slipper slope, I have wanted something like this for a while, but not a strong want...
More in terms of linking things from example tables than actual steps.
So maybe have speciality linking functions would be more handy...

The code is quite clean, but before I sign off I do want to pull it and run against my work code base too. Hopefully tomorrow, if not, over the weekend.

File: sphinx_gherkindoc/cli.py [hunk 1]
```diff
@@ -51,19 +51,26 @@ def process_args(
     non_empty_dirs: Set[pathlib.Path] = set()
 
     get_url_from_tag = None
+    get_url_from_step = None
     dir_display_name_converter = None
     # Set parsers once and pass along where they are needed.
     for entry_point in pkg_resources.iter_entry_points("parsers"):
-        if entry_point.name == "url":
+        if entry_point.name in ("url", "tag_url"):
```
Comment (rbcasperson) [inline sphinx_gherkindoc/cli.py:58]: Maybe just note the legacy bit here in the code since this is where it is relevant.

**Comment by bradsbrown** [comment]
Travis passes, but isn't closing the "hook" back to GH for some reason. Merging with 3 approvals and [passing CI](https://travis-ci.org/github/jolly-good-toolbelt/sphinx_gherkindoc/builds/670727486?utm_medium=notification&utm_source=slack)

**Pull request merged**